server/tests/configuration.test.ts

```diff
@@ -1,35 +1,105 @@
-import { decodeConfiguration } from '../src/configuration'
+import { Octokit } from 'probot'
+import mls from 'multilines'
+import { getConfig } from '../src/configuration'
+import { Sources } from '../src/sources'
+import { getMockSources } from './__mock__/sources'
 
 describe('configuration:', () => {
-  test('errors on incorrect', () => {
-    const config = {
-      starters: [
-        {
-          name: 'starter',
-          path: '/templates/starter',
-        },
-        {
-          faulty: '',
+  describe('getConfig:', () => {
+    let sources: Sources = getMockSources()
+    let github: Octokit
+    let getContents: jest.Mock
+
+    const owner = 'maticzav'
+    const repo = 'emma-cli'
+
+    beforeAll(() => {
+      sources = getMockSources()
+    })
+
+    beforeEach(() => {
+      github = new Octokit()
+      getContents = jest.fn()
+      github.repos.getContents = getContents as any
+    })
+
+    test('should return null if occur some error', async () => {
+      getContents.mockRejectedValueOnce(new Error('exception'))
+
+      const result = await getConfig(sources)(github, owner, repo)
+
+      expect(result).toBeNull()
+      expect(getContents).toBeCalledTimes(1)
+    })
+
+    test('should return null if repo is not a file', async () => {
```

Comment:
I think it might be better to fill the code with `/* istanbul ignore next */` (i.e. coverage fillers) than to cover every possible case there exists. I suppose you are quite clever so you can decide which cases require testing and which are better off skipped.